internal/backend/invite_request.go

```diff
@@ -0,0 +1,267 @@
+package backend
+
+import (
+	"context"
+	"io"
+
+	"github.com/nedrocks/delphisbe/internal/util"
+	"go.uber.org/multierr"
+
+	"github.com/nedrocks/delphisbe/graph/model"
+	"github.com/nedrocks/delphisbe/internal/datastore"
+	"github.com/sirupsen/logrus"
+)
+
+func (d *delphisBackend) GetDiscussionInviteByID(ctx context.Context, id string) (*model.DiscussionInvite, error) {
+	return d.db.GetDiscussionInviteByID(ctx, id)
+}
+
+func (d *delphisBackend) GetDiscussionRequestAccessByID(ctx context.Context, id string) (*model.DiscussionAccessRequest, error) {
+	return d.db.GetDiscussionRequestAccessByID(ctx, id)
+}
+
+func (d *delphisBackend) GetDiscussionInvitesByUserIDAndStatus(ctx context.Context, userID string, status model.InviteRequestStatus) ([]*model.DiscussionInvite, error) {
+	iter := d.db.GetDiscussionInvitesByUserIDAndStatus(ctx, userID, status)
+	return d.iterToDiscussionInvites(ctx, iter)
+}
+
+func (d *delphisBackend) GetSentDiscussionInvitesByUserID(ctx context.Context, userID string) ([]*model.DiscussionInvite, error) {
+	iter := d.db.GetSentDiscussionInvitesByUserID(ctx, userID)
+	return d.iterToDiscussionInvites(ctx, iter)
+}
+
+func (d *delphisBackend) GetDiscussionAccessRequestsByDiscussionID(ctx context.Context, discussionID string) ([]*model.DiscussionAccessRequest, error) {
+	iter := d.db.GetDiscussionAccessRequestsByDiscussionID(ctx, discussionID)
+	return d.iterToDiscussionAccessRequests(ctx, iter)
+}
+
+func (d *delphisBackend) GetSentDiscussionAccessRequestsByUserID(ctx context.Context, userID string) ([]*model.DiscussionAccessRequest, error) {
+	iter := d.db.GetDiscussionAccessRequestsByDiscussionID(ctx, userID)
+	return d.iterToDiscussionAccessRequests(ctx, iter)
+}
+
+func (d *delphisBackend) InviteUserToDiscussion(ctx context.Context, userID, discussionID, inviteeID string) (*model.DiscussionInvite, error) {
```

Comment:
I have a feeling we will want to invite users by twitter tag more often than by user ID (Although both are probably important). Should we support the former here?

Reply:
let's discuss this more. tracking here: https://github.com/nedrocks/delphisbe/issues/35